ddl/table.go

```diff
@@ -213,6 +213,26 @@ func onTruncateTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
+
+	var oldPartitionIDs []int64
+	if tblInfo.GetPartitionInfo() != nil {
+		oldPartitionIDs = getPartitionIDs(tblInfo)
+		for _, def := range tblInfo.Partition.Definitions {
+			var pid int64
+			genID := func(txn kv.Transaction) error {
```

Comment:
You can move ddl.genGlobalID to ddlCtx.genGlobalID, then you can call `ddlCtx.genGlobalID` directly.  Don't keep the same logic code in two place.

Reply:
@winkyao Move the code too much, do it alone with another `pr`.